### ISSUE TITLE
[Draft] .Net Core 3 support: helpers for newly added types/interfaces

### ISSUE THREAD

**Issue by ig-sinicyn** [rsdn/CodeJam#94]
Will update with things planned for CJ 3.1. Feel free to suggest  additions to the list:)

* Add helpers for IAsyncDisposable
* Add support for IAsyncDisposable to AsyncLock.

**Comment by NN---** [comment]
Do you mean something similar to https://github.com/rsdn/CodeJam/blob/master/CodeJam.Main/Disposable.cs ?

**Comment by ig-sinicyn** [comment]
Yep. There's a branch for async disposable but the code is in complete.
The issue meant to be an umbrella for netcore3-related features. But it turns out we are out of ideas:)

**Comment by NN---** [comment]
Why can't you add what you already have ?

**Comment by ig-sinicyn** [comment]
The branch is pushed to repo already. I think I will complete it on next weekend.

**Comment by NN---** [comment]
I mean why it is not in master :)

**Comment by ig-sinicyn** [comment]
Because the feature is not ready yet and I do not want to add unstable code to the master.

**Comment by NN---** [comment]
Any update for this?:)
